src/praisonai/praisonai/cli_backends/registry.py

```
@@ -47,12 +77,15 @@ def resolve_cli_backend(
-            raise ValueError(f"Unknown CLI backend: {backend_id}. Available: {available}")
-        
-        factory = _REGISTRY[backend_id]
+    registry = _get_cli_backend_registry()
+    
+    try:
+        # Get the factory loader and create the factory
+        factory_loader = registry.resolve(backend_id)
+        factory = factory_loader()
+    except ValueError:
+        available = registry.list_names()
+        raise ValueError(f"Unknown CLI backend: {backend_id}. Available: {available}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)